flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/trait/UpdateKind.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.plan.trait;
+
+/**
+ * Lists all kinds of {@link ModifyKind#UPDATE} operation.
+ */
+public enum UpdateKind {
+
+	/**
+	 * No update operation.
+	 */
+	NO_UPDATE,
+
+	/**
+	 * This kind indicates that operators should emit update changes just as a row of
+	 * {@code RowKind#UPDATE_AFTER}.
+	 */
+	ONLY_UPDATE_AFTER,
+
+	/**
+	 * This kind indicates that operators should emit update changes in the way that
+	 * a row of {@code RowKind#UPDATE_BEFORE} and a row of {@code RowKind#UPDATE_AFTER} together.
+	 */
+	BEFORE_AND_AFTER;
+
+	/**
+	 * Return a new UpdateKind which merges this UpdateKind and the other UpdateKind.
+	 */
+	public UpdateKind merge(UpdateKind that) {
```

Comment:
It's hard to imagine & understand why this class support `merge`, what should be the expected behavior?

Reply:
Children of Union may have different UpdateKind. This is used to "union" children's UpdateKind. I was thinking "union" is not fit for this enum class, because it is not a bitset. 

I moved this logic into `FlinkChangelogModeInferenceProgram` as it is only used by StreamExecUnion.